mailing-list/main.py

```diff
@@ -211,14 +214,45 @@ def parse_dumps():
 
 
 def index_documents(docs):
-    for doc in docs:
-
-        resp = document_view(index_name=INDEX, doc_id=doc['id'])
-        if not resp:
-            _ = document_add(index_name=INDEX, doc=doc, doc_id=doc['id'])
-            logger.success(f'Successfully added! ID: {doc["id"]}')
-        else:
-            logger.info(f"Document already exist! ID: {doc['id']}")
+    inserted_ids = set()
+    updated_ids = set()
+    no_changes_ids = set()
+    error_occurred = False
+    error_message = "---"
+    try:
+        for doc in tqdm(docs):
+            try:
+                res = upsert_document(index_name=os.getenv('INDEX'), doc_id=doc['id'], doc_body=doc)
+                if res['result'] == 'created':
+                    inserted_ids.add(res['_id'])
+                elif res['result'] == 'updated':
+                    updated_ids.add(res['_id'])
+                elif res['result'] == 'noop':
+                    no_changes_ids.add(res['_id'])
+
+            except Exception as e:
+                logger.error(f"Error upserting document ID-{doc['id']}: {e}")
+                logger.warning(doc)
+
+    except Exception as main_e:
+        logger.error(f"Main Error: {main_e}")
+        error_occurred = True
+        error_message = str(main_e)
```

Comment:
The existence of a non-empty `error_message` means that an `error_occurred`, right? It seems that the `error_occurred` variable is redundant.